BoundedKnapsack/ReferenceImplementation.qs

```diff
@@ -0,0 +1,428 @@
+//////////////////////////////////////////////////////////////////////
+// This file contains reference solutions to all tasks.
+// The tasks themselves can be found in Tasks.qs file.
+// We recommend that you try to solve the tasks yourself first,
+// but feel free to look up the solution if you get stuck.
+//////////////////////////////////////////////////////////////////////
+
+namespace Quantum.Kata.BoundedKnapsack {
+
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Arithmetic;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Measurement;
+
+
+    //////////////////////////////////////////////////////////////////
+    // Part I. 0-1 Knapsack Problem
+    //////////////////////////////////////////////////////////////////
+	
+
+	// Task 1.1. Read combination from a register
+	operation MeasureCombination_01_Reference (register : Qubit[]) : Bool[] {
+		return ResultArrayAsBoolArray(MultiM(register));
+	}
+	
+
+	// Task 1.2. Calculate Total Weight or Profit
+	operation CalculateTotalWeightOrProfit_01_Reference (n : Int, values : Int[], register : Qubit[], total : Qubit[]) : Unit is Adj+Ctl{
+		// Each qubit in xs determines whether the corresponding value is added.
+		// This process is implemented with a control from xs[i].
+		let TotalLE = LittleEndian(total);
+		for (i in 0..n-1){
+			Controlled IncrementByInteger([register[i]], (values[i], TotalLE));
+		}

```

Comment:
```suggestion
		for ((control, value) in Zip(register, values)) {
			Controlled IncrementByInteger([control], (value, TotalLE));
		}
```

Alternative to avoid the loop counting variable.